### PR TITLE
Hotfix/url bugs and rush threshold

### PR DESCRIPTION
**Pull request by wderocco8**
## What does this PR do?

1. Adds logic to decide if rush has attended the appropriate number of events for an interview
2. Fixed bug when opening URLs that are not prefixed by "http" or "https"

## Type of change

- [x] Fix: Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor: Any code refactoring
- [ ] Chore: technical debt, workflow improvements
- [ ] Feature: New feature (non-breaking change which adds functionality)
- [ ] Documentation: This change requires a documentation update

### CODE REVIEW

File: utils/getRushThreshold.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+export const getRushThreshold = (events: { [key: string]: boolean }) => {
+  // Check if at least one mandatory event was attended
+  const attendedMandatory = mandatoryEvents.some((event) => events[event]);
+
+  // Count how many remaining events were attended
+  const attendedRemaining = remainingEvents.filter((event) => events[event]).length;
+
+  // Check if the rush threshold is met: at least 1 mandatory event and 2 other events
+  return attendedMandatory && attendedRemaining >= 2;
```
Comment (jinyoungbang) [inline utils/getRushThreshold.ts:14]: Can we make the count `2` is a constant? Additionally, you use that logic on both functions so maybe you can abstract this by creating a helper function like `isRushThresholdMet(...)` - verbosity in this case would help understand the functionality better.

File: utils/getRushThreshold.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { AnalyticsEvent } from "@/types/admin/events";
+
+const mandatoryEvents = ["Info Session 1", "Info Session 2"]
+const remainingEvents = ["Professional Panel", "Resume Night", "Social Event"]
+
+export const getRushThreshold = (events: { [key: string]: boolean }) => {
+  // Check if at least one mandatory event was attended
+  const attendedMandatory = mandatoryEvents.some((event) => events[event]);
```
Comment (jinyoungbang) [inline utils/getRushThreshold.ts:8]: nit - `attendedMandatoryEvent` would be a better name

File: utils/getRushThreshold.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  // Count how many remaining events were attended
+  const attendedRemaining = remainingEvents.filter((event) => events[event]).length;
```
Comment (jinyoungbang) [inline utils/getRushThreshold.ts:11]: nit - `numAttendedRemainingEvents`? or maybe a better name would be better.

**Review by jinyoungbang — CHANGES_REQUESTED** [review]

File: utils/getRushThreshold.ts (hunk 1 above)
Comment (wderocco8) [reply utils/getRushThreshold.ts:14]: Completely agree on both. I'll update `2` to be a constant, but Imma come back to the abstraction on a separate PR bc I'm a bit swamped rn lol (and I wanna fix the analytics processing in the backend anyways)

**Pull request merged**